pact/Pact/Core/Errors.hs

```diff
@@ -1061,6 +1095,586 @@ deriveConstrInfo ''EvalError
+  NestedDefPactDoubleExecution ps -> thsep
+    ["Requested nested DefPact double execution:", "DefPactId:" , tDpId (_psDefPactId ps)]
+  MultipleOrNestedDefPactExecFound pe -> thsep
+    ["DefPact execution context already in the environment: DefPactId:", tDpId (_peDefPactId pe)]
+  DefPactStepHasNoRollback ps -> thsep
+    ["Step has no rollback:", "DefPactId:", tDpId (_psDefPactId ps)]
+  DefPactStepNotInEnvironment -> "No DefPactStep in the environment"
+  NoDefPactIdAndExecEnvSupplied -> "No DefPactId or execution environment supplied"
+  DefPactRollbackMismatch ps pe -> thsep
+    [ "Rollback missmatch in DefPactStep and DefPact execution environment:"
+    , "DefPactId:", tDpId (_psDefPactId ps)
+    , "step rollback:", tBool (_psRollback ps)
+    , "DefPactExec rollback:", tBool (_peStepHasRollback pe)
+    ]
+  DefPactStepMismatch ps pe -> thsep
+    [ "Step missmatch in DefPactStep and DefPact execution environment:"
+    , "DefPactId:", tDpId (_psDefPactId ps)
+    , "step:", tInt (_psStep ps)
+    , "DefPactExec step:", tInt (_peStep pe + 1)
+    ]
+  EnforcePactVersionFailure min' max' -> thsep
+    [ "Enforce pact-version failed:"
+    , "Current Version:", T.pack (V.showVersion PI.version)
+    , "Minimum Version:", T.pack (V.showVersion min')
+    , maybe mempty (\v -> "Maximum Version: " <> T.pack (V.showVersion v)) max'
+    ]
+  EnforcePactVersionParseFailure str -> thsep
+    [ "Enforce pact-version failed:"
+    , "Could not parse", str, ", expect list of dot-separated integers"
+    ]
+  InvalidManagedCap fqn ->
+    thsep ["Install capability error: capability is not managed and cannot be installed:", tFqn fqn]
+  CapNotInstalled cap ->
+    thsep
+      [ "Capability not installed:", renderQualName (_ctName cap)
+      , ". Check that the capabilty is in the sigs or the arguments are correct"]
+  CapAlreadyInstalled cap ->
+    thsep ["Capability already installed:", renderQualName (_ctName cap)]
+  ModuleMemberDoesNotExist fqn ->
+    thsep ["Module member does not exist", tFqn fqn]
+  NoSuchKeySet ksn ->
+    thsep ["Cannot find keyset in database:", renderKeySetName ksn]
+  CannotUpgradeInterface ifn ->
+    thsep ["Interface cannot be upgraded:", renderModuleName ifn]
+  DbOpFailure dbe ->
+    thsep ["Error during database operation:", dbOpErrorToBoundedText' dbe]
+  DynNameIsNotModRef n ->
+    thsep ["Attempted to use", abbrevText 25 n, "as dynamic name, but it is not a modref"]
+  ModuleDoesNotExist m ->
+    thsep ["Cannot find module:", renderModuleName m]
+  ExpectedModule mn ->
+    thsep ["Expected module, found interface:", renderModuleName mn]
+  HashNotBlessed mn hs  ->
+    thsep
+      ["Execution aborted, hash not blessed for module"
+      , renderModuleName mn
+      , "with hash"
+      , moduleHashToText hs]
+  CannotApplyPartialClosure ->
+    "Attempted to apply a partially applied closure outside of native callsite"
```

Comment:
Maybe "function" is better than "closure" here for users? You could also add "Pact does not support user-defined higher-order functions."